x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -54,6 +49,7 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 
 	// construct desired unstaking amount from host zone
 	stDenom := types.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
+	// ASSUMPTION (CHECK ME): it doesn't matter that RRs can change _within_ an epoch (due to LSM)
```

Comment:
I double checked the code base and the only times we use the RR are in liquid stake and redeem. I believe this implies that in other places we just use the total number of tokens, and it wouldn't matter if that collection of tokens was built up from different rates.

Similarly, the only times we modify NativeTokenAmount are in redeem and claim, and the only times we modify StTokenAmount are in redeem and undelegate.

Additionally, in v18, if we end up doing the st → native conversion at the time that the unbonding is initiated (instead of during redemption), we will no longer even have to calculate this here

Reply:
I think we should always prefer to use the most up-to-date redemption rate because it will more accurately reflect ground truth NativeTokenAmount vs stTokenAmount. 

It's desirable to unstake an amount of tokens commensurate with the most up-to-date RR.

Now that the RR changes _within_ an epoch, that's better than when the RR updated only epochly.

Reply:
One nit comment, the RRs can change _within_ an epoch outside of LSM too, because the URR are keyed daily and the redemption rate changes based on `stride_epochs` (i.e. 6 hours)

Reply:
Good point @shellvish. I had thought we were keying on stride_epoch, surprised we're keying on day.

In any case, all of the above comments make sense to me! Thanks for sanity checking this approach.

Reply:
Good callout @shellvish, I forgot they were key'd daily as well! 

Marking this comment resolved cause it appears we're all in agreement!